tests/Unit/SymfonyCache/UserContextSubscriberTest.php

```diff
@@ -177,6 +177,58 @@ public function testUserHashUserWithSession($arg, $options)
         $this->assertSame($expectedContextHash, $request->headers->get($options['user_hash_header']));
     }
 
+    /**
+     * @dataProvider provideConfigOptions
+     */
+    public function testUserHashUserWithAuthorizationHeader($arg, $options)
+    {
+        $userContextSubscriber = new UserContextSubscriber($arg);
+
+        $catch = true;
+        $request = Request::create('/foo', 'GET', array(), array(), array(), array('HTTP_AUTHORIZATION' => 'foo'));
+
+        $hashRequest = Request::create($options['user_hash_uri'], $options['user_hash_method'], array(), array(), array(), $request->server->all());
+        $hashRequest->attributes->set('internalRequest', true);
+        $hashRequest->headers->set('Accept', $options['user_hash_accept_header']);
+
+        // Ensure request properties have been filled up.
+        $hashRequest->getPathInfo();
+        $hashRequest->getMethod();
+
+        $expectedContextHash = 'my_generated_hash';
+        // Just avoid the response to modify the request object, otherwise it's impossible to test objects equality.
+        /** @var \Symfony\Component\HttpFoundation\Response|\PHPUnit_Framework_MockObject_MockObject $hashResponse */
+        $hashResponse = $this->getMockBuilder('\Symfony\Component\HttpFoundation\Response')
+            ->setMethods(array('prepare'))
+            ->getMock();
+        $hashResponse->headers->set($options['user_hash_header'], $expectedContextHash );
+
+        $that = $this;
+        $this->kernel
+            ->expects($this->once())
+            ->method('handle')
+            ->with(
+                $this->callback(function (Request $request) use ($that, $hashRequest) {
+                    // we need to call some methods to get the internal fields initialized
+                    $request->getMethod();
+                    $request->getPathInfo();
+                    $that->assertEquals($hashRequest, $request);
+
+                    return true;
+                }),
+                $catch
```

Comment:
To be honest: I have no idea. I just tried to adjust the very complicated existing test from the session approach.